diff --git a/mimemail.inc b/mimemail.inc
index 938ceb3..6a406a6 100644
--- a/mimemail.inc
+++ b/mimemail.inc
@@ -77,7 +77,9 @@ function mimemail_headers($headers, $from = NULL) {
   // Run all headers through mime_header_encode() to convert non-ascii
   // characters to an rfc compliant string, similar to drupal_mail().
   foreach ($headers as $key => $value) {
-    $headers[$key] = mime_header_encode($value);
+    if (preg_match('/\=\?UTF-8\?B\?.*\?\=\n$/', $value) == FALSE) {
+      $headers[$key] = mime_header_encode($value);
+    }
   }
 
   return $headers;
